### PR TITLE
Draft: Eliminate the order class in post order #2469

### LINKED ISSUE #2469: chore: Eliminate the order class in post order
### Background
Since all the orders are limit orders, eliminate the order class from the post order.

Keep the order class in the database as an internal class with: `limit` and `market`. By doing this, we can easily count the number of real `limit` orders in order to limit the number of orders per user.

### Details
It has to be done:
- remove the order class in post order
- classify properly internally if an order is `market` or `limit`
- eliminate `liquidity` class order from the database (migration needed to move them as `limit` order)
- change the way we count of "in-market` order to use the database class

### Acceptance criteria
The changes are backwards compatible in behavior, but the API contract is changed. All the current tests must pass.

### PR DESCRIPTION
**Pull request by Mateo-mro**
# Description
Since all the user orders are limit orders, eliminate the order class from the post order.

# Changes
- eliminate the order class from the `PreOrder` (since all the user orders should be `limit` orders)
- decide if an user order is `limit` or `market` based on its quote value on order submit time
- count the limit of user limits orders based on the order class in the database (since now it should really match its real order)
- keep `liquidity`, it is complex to remove it and it should be done in follow up MRs.

## How to test
1. unit test
2. e2e test
## Related Issues

Fixes #2469

### CODE REVIEW

File: crates/orderbook/src/database/orders.rs [hunk 1]
```diff
@@ -367,33 +364,12 @@ impl LimitOrderCounting for Postgres {
             .start_timer();
 
         let mut ex = self.pool.acquire().await?;
-        Ok(database::orders::user_orders_with_quote(
```
[the remark this answers is not in the document]
Comment (Mateo-mro) [reply crates/orderbook/src/database/orders.rs:370]: @MartinquaXD the orders which are real "in-market" orders should be executed relatively soon (minutes I believe), leaving the database populated only with real `limit` orders very soon. If we don't do a migration, the risk is to hitting the too many limit orders because we may count "in-market" orders as limit orders, but the question is for how long will those orders live?
Otherwise, we would need to do a migration which goes through all the orders, fetch the quote, and check if they are "in-market" or not to reclassify them, which I can also do, I was just wondering if we would really need it. WDYT?

[the remark this answers is not in the document]
Comment (MartinquaXD) [reply crates/orderbook/src/database/orders.rs:370]: While I think you are right that we'd not run into order counting issues for long it would still be good to retroactively label orders correctly so we have a consistent state in the DB.
Otherwise people will have to remember that the meaning of the label changed at some point when they look into some old order (which can happen if we do some historic analysis).

File: crates/shared/src/order_validation.rs [hunk 2]
```diff
@@ -1647,70 +1582,6 @@ mod tests {
...
-            Arc::new(order_validation::banned::Users::none()),
-            OrderValidPeriodConfiguration::any(),
-            false,
-            Arc::new(bad_token_detector),
-            dummy_contract!(HooksTrampoline, [0xcf; 20]),
-            Arc::new(order_quoter),
-            Arc::new(balance_fetcher),
-            Arc::new(MockSignatureValidating::new()),
-            Arc::new(limit_order_counter),
-            0,
```
[the remark this answers is not in the document]
Comment (Mateo-mro) [reply crates/shared/src/order_validation.rs:1684]: @MartinquaXD I investigated this test, the issue is that the limit for user limit orders is 0, and the order it is trying to place is a limit order because it is not "in-market", so the test fails. I thought this test is superfluous since we have these two tests:
- https://github.com/cowprotocol/services/pull/2496/files#diff-1ff024ca2dfa09435b86bbecf6919e50319d070f04f71ed834341d5d254b3c15L1451
- https://github.com/cowprotocol/services/pull/2496/files#diff-1ff024ca2dfa09435b86bbecf6919e50319d070f04f71ed834341d5d254b3c15L1532

File: crates/orderbook/src/database/orders.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (Mateo-mro) [reply crates/orderbook/src/database/orders.rs:370]: oh, that makes a lot of sense! got it! perfect, I will prepare a migration for it then! thank you!

File: crates/e2e/tests/e2e/limit_orders.rs [hunk 3]
```diff
@@ -364,9 +362,9 @@ async fn mixed_limit_and_market_orders_test(web3: Web3) {
 
     let order_a = OrderCreation {
         sell_token: token_a.address(),
-        sell_amount: to_wei(10),
+        sell_amount: to_wei(1),
         buy_token: token_b.address(),
-        buy_amount: to_wei(5),
+        buy_amount: to_wei(1),
```
Comment (squadgazzz) [inline crates/e2e/tests/e2e/limit_orders.rs:367]: Why this change is required?
